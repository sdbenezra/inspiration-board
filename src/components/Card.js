import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {
  constructor(props) {
    super(props)
  }
  render() {
    console.log(`Text is: ${this.props.text}`);
    console.log(`Emoji is: ${this.props.emoji}`);

    return (
      <div className="card">
        <div className="card__content">
          <p>{this.props.text}</p>
          <p>{emoji.getUnicode(`${this.props.emoji}`)}</p>
          </div>
      </div>
    )
  }
}

Card.propTypes = {
  text: PropTypes.text,
  emoji: PropTypes.text,
};

export default Card;
