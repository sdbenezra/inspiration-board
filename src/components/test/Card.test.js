import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  test('that it matches an existing snapshot', () => {
    // First Mount the Component in the testing DOM
    // Arrange
    const wrapper = shallow(
       <Card
          text = 'text'
          emoji = 'emoji'
          />
    );

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();
  });
});
