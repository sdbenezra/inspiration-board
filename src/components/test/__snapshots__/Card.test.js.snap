// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches an existing snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    emoji="emoji"
    text="text"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card__content"
>
        <p
                className="card__content-text"
        >
                text
        </p>
        <p
                className="card__content-emoji"
        />
        <button
                className="card__delete"
                onClick={[Function]}
        >
                Delete
        </button>
</div>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="card__content-text"
>
            text
</p>,
          <p
            className="card__content-emoji"
/>,
          <button
            className="card__delete"
            onClick={[Function]}
>
            Delete
</button>,
        ],
        "className": "card__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "text",
            "className": "card__content-text",
          },
          "ref": null,
          "rendered": "text",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "card__content-emoji",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "className": "card__delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card__content"
>
          <p
                    className="card__content-text"
          >
                    text
          </p>
          <p
                    className="card__content-emoji"
          />
          <button
                    className="card__delete"
                    onClick={[Function]}
          >
                    Delete
          </button>
</div>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="card__content-text"
>
              text
</p>,
            <p
              className="card__content-emoji"
/>,
            <button
              className="card__delete"
              onClick={[Function]}
>
              Delete
</button>,
          ],
          "className": "card__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "text",
              "className": "card__content-text",
            },
            "ref": null,
            "rendered": "text",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "card__content-emoji",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Delete",
              "className": "card__delete",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Delete",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
